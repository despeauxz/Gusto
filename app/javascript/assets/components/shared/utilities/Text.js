import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

const Text = ({ color, variant, size, font, children }) => {
    return (
        <span className={classnames({ [`text-${color}-${variant} text-${size} font-${font}`]: true })}>
      {children}
    </span>
    );
};

Text.propTypes = {
    color: PropTypes.string.isRequired,
    variant: PropTypes.number.isRequired,
    size: PropTypes.oneOf(['xs', 'sm', 'base', 'lg', 'xl', '2xl', '3xl', '4xl', '5xl', '6xl']),
    font: PropTypes.oneOf(['sans', 'serif', 'mono']),
    children: PropTypes.oneOf([
        PropTypes.string,
        PropTypes.element
    ]).isRequired
};

Text.defaultProps = {
    size: "base",
    font: "sans"
};

export default Text;
