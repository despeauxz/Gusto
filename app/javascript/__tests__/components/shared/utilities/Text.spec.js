/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import Text from '../../../../assets/components/shared/utilities/Text';


const props = {
    color: "blue",
    variant: 600,
    size: "sm",
    font: "sans",
};

describe('<Text />', () => {
    it('render the Text component', () => {
        const shallowWrapper = shallow(<Text {...props}>
            Hello World
        </Text>);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('Text')).toBeTruthy();
    });
}); 
