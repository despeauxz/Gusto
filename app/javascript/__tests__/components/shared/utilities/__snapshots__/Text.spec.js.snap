// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> render the Text component 1`] = `
<span
  className="text-blue-600 text-sm font-sans"
>
  Hello World
</span>
`;
